data/network/src/commonMain/kotlin/social/androiddev/common/network/MastodonApiKtor.kt

```diff
@@ -36,29 +40,62 @@ internal class MastodonApiKtor(
         }
     }
 
+    override suspend fun createAccessToken(
+        domain: String,
+        clientId: String,
+        clientSecret: String,
+        redirectUri: String,
+        grantType: String,
+        code: String,
+        scope: String
+    ): Result<Token> {
+        return runCatchingIgnoreCancelled<Token> {
+            httpClient.post {
+                url {
+                    host = domain
+                    path("/oauth/token")
+                }
+                contentType(ContentType.Application.Json)
+                setBody(
+                    CreateAccessTokenBody(
+                        scope = scope,
+                        code = code,
+                        clientId = clientId,
+                        clientSecret = clientSecret,
+                        redirectUri = redirectUri,
+                        grantType = grantType,
+                    )
+                )
+            }.body()
+        }.onFailure { t ->
+            t.printStackTrace()
+        }
+    }
+
     override suspend fun createApplication(
         domain: String,
         clientName: String,
         redirectUris: String,
         scopes: String,
         website: String?
     ): Result<NewOauthApplication> {
-        return runCatchingIgnoreCancelled {
+        return runCatchingIgnoreCancelled<NewOauthApplication> {
             httpClient.post {
                 url {
-                    protocol = URLProtocol.HTTPS
                     host = domain
                     path("/api/v1/apps")
                 }
-                formData {
-                    append("client_name", clientName)
-                    append("redirect_uris", redirectUris)
-                    append("scopes", scopes)
-                    if (website != null) {
-                        append("website", website)
-                    }
-                }
+                contentType(ContentType.Application.Json)
+                setBody(
+                    CreateApplicationBody(
+                        scopes = scopes,
+                        clientName = clientName,
+                        redirectUris = redirectUris
+                    )
+                )
             }.body()
+        }.onFailure { t ->
+            t.printStackTrace()
```

Comment:
same - log or rethrow

Reply:
btw have we settled on a logger? Is there a KMP one?

Reply:
Haven't settled on a Logger yet, but would be wise to get that setup sooner than later so we can remove replace all `printStackTrace()` calls and add better logging throughout